internal/workload/podspec_updates.go

```diff
@@ -330,36 +334,57 @@ func (s *updateState) addPort(p int32, containerName string, n types.NamespacedN
 
 	if mp == nil {
 		mp = &managedPort{
-			Instance:       dbInst(n.Namespace, n.Name, connectionString),
-			Port:           p,
-			OriginalValues: map[string]int32{},
+			Instance: dbInst(n.Namespace, n.Name, connectionString),
+			Port:     p,
 		}
 		s.mods.Ports = append(s.mods.Ports, mp)
 	}
-	if containerName != "" && !strings.HasPrefix(containerName, ContainerPrefix) {
-		mp.OriginalValues[containerName] = p
-	}
+}
 
+func (s *updateState) addProxyContainerEnvVar(p *cloudsqlapi.AuthProxyWorkload, k, v string) {
+	s.addEnvVar(p, managedEnvVar{
+		Instance:             dbInst(p.Namespace, p.Name, ""),
```

Comment:
Again the empty string confuses me here.

Reply:
Same here.